[VL] Error when creating shuffle file: open with O_CREAT or O_TMPFILE in second argument needs 3 arguments

Error:

```
In file included from /usr/include/fcntl.h:301,
                 from /opt/gluten/dev/vcpkg/vcpkg_installed/x64-linux-avx/include/boost/stacktrace/detail/safe_dump_posix.ipp:18,
                 from /opt/gluten/dev/vcpkg/vcpkg_installed/x64-linux-avx/include/boost/stacktrace/safe_dump_to.hpp:212,
                 from /opt/gluten/dev/vcpkg/vcpkg_installed/x64-linux-avx/include/boost/stacktrace/frame.hpp:18,
                 from /opt/gluten/dev/vcpkg/vcpkg_installed/x64-linux-avx/include/boost/stacktrace.hpp:15,
                 from /opt/gluten/cpp/core/shuffle/LocalPartitionWriter.cc:22:
In function ‘int open(const char*, int, ...)’,
    inlined from ‘arrow::Result<std::shared_ptr<arrow::io::OutputStream> > gluten::LocalPartitionWriter::openFile(const string&)’ at /opt/gluten/cpp/core/shuffle/LocalPartitionWriter.cc:390:17:
/usr/include/x86_64-linux-gnu/bits/fcntl2.h:50:31: error: call to ‘__open_missing_mode’ declared with attribute error: open with O_CREAT or O_TMPFILE in second argument needs 3 arguments
   50 |           __open_missing_mode ();
      |           ~~~~~~~~~~~~~~~~~~~~^~
```

The error appears on ubuntu 20.04.

Could be related to https://github.com/apache/incubator-gluten/pull/9156.

--- a/cpp/core/shuffle/LocalPartitionWriter.cc
+++ b/cpp/core/shuffle/LocalPartitionWriter.cc
@@ -387,7 +387,7 @@ std::string LocalPartitionWriter::nextSpilledFileDir() {
 
 arrow::Result<std::shared_ptr<arrow::io::OutputStream>> LocalPartitionWriter::openFile(const std::string& file) {
   std::shared_ptr<arrow::io::FileOutputStream> fout;
-  auto fd = open(file.c_str(), O_WRONLY | O_CREAT | O_TRUNC);
+  auto fd = open(file.c_str(), O_WRONLY | O_CREAT | O_TRUNC, 0000);
   // Set the shuffle file permissions to 0644 to keep it consistent with the permissions of
   // the built-in shuffler manager in Spark.
   fchmod(fd, 0644);

